hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -537,14 +537,14 @@ func allOfHandler(ctx context.Context, scanner *SchemaScanner, schema *gojsonsch
 	var handleType func(properties []*astmodel.PropertyDefinition, st astmodel.Type) ([]*astmodel.PropertyDefinition, error)
 	handleType = func(properties []*astmodel.PropertyDefinition, st astmodel.Type) ([]*astmodel.PropertyDefinition, error) {
 		switch concreteType := st.(type) {
-		case *astmodel.StructType:
+		case *astmodel.ObjectType:
 			// if it's a struct type get all its properties:
 			properties = append(properties, concreteType.Properties()...)
 
 		case *astmodel.TypeName:
 			// TODO: need to check if this is a reference to a struct type or not
 			if def, ok := scanner.findTypeDefinition(concreteType); ok {
-				if structDef, ok := def.(*astmodel.StructDefinition); ok {
+				if structDef, ok := def.(*astmodel.ObjectDefinition); ok {
```

Comment:
"struct" used here but don't worry about changing it because I'm editing this again later :)